userinterface/Views/MainWindow.axaml.cs

```diff
@@ -1,30 +1,75 @@
-﻿using Avalonia;
+using Avalonia;
 using Avalonia.Controls;
 using Avalonia.Controls.Primitives;
 using Avalonia.Interactivity;
 using Avalonia.Styling;
+using Microsoft.Extensions.DependencyInjection;
+using System;
 using System.Threading.Tasks;
+using userinterface.Models;
 using userinterface.Services;
 using userinterface.ViewModels;
+using userinterface.Views.Controls;
 
 namespace userinterface.Views;
 
 public partial class MainWindow : Window
 {
     private Button? ApplyButtonControl;
     private ProgressBar? LoadingProgressBar;
-    private readonly INotificationService notificationService;
-    private readonly IModalService modalService;
 
-    public MainWindow(INotificationService notificationService, IModalService modalService)
+    public MainWindow()
     {
         InitializeComponent();
-        this.notificationService = notificationService;
-        this.modalService = modalService;
+
+        InitializeControls();
         UpdateThemeToggleButton();
-        UpdateSelectedButton("Devices");
+        UpdateSelectedButton(NavigationPage.Devices);
+    }
+
+    private INotificationService NotificationService =>
+        App.Services!.GetRequiredService<INotificationService>();
+
+    private void InitializeControls()
+    {
         ApplyButtonControl = this.FindControl<Button>("ApplyButton");
         LoadingProgressBar = this.FindControl<ProgressBar>("LoadingProgress");
+
+        // Subscribe to click events
+        if (ApplyButtonControl != null)
+        {
+            ApplyButtonControl.Click += ApplyButtonHandler;
+        }
+
+        var settingsButton = this.FindControl<Button>("SettingsButton");
```

Comment:
I feel like this would be cleaner if we wrapped FindControl<T>(string) with TryFindControl<T>(string, out T)